File: Top 100 Codes/05-Operations on Strings/13-Count the sum of numbers in a string/Countthesumofnumbersinastring.py
def sum_of_numbers_in_string(s):
    total_sum = 0
    num = 0

    for ch in s:
        if ch.isdigit():
            num = num * 10 + int(ch)  # Form the number
        else:
            total_sum += num  # Add the formed number to sum
            num = 0  # Reset for next number

    total_sum += num  # Add the last number (if any)
    return total_sum

# Driver code
input_string = input("Enter a string: ")
print("Sum of numbers in the string:", sum_of_numbers_in_string(input_string))

File: Top 100 Codes/05-Operations on Strings/13-Count the sum of numbers in a string/Countthesumofnumbersinastring.cpp
#include <iostream>
#include <cctype>
using namespace std;

// Function to calculate sum of numbers in a string
int sumOfNumbersInString(string str) {
    int sum = 0, num = 0;

    for (int i = 0; i < str.length(); i++) {
        if (isdigit(str[i])) {
            num = num * 10 + (str[i] - '0'); // Form the number
        } else {
            sum += num; // Add the formed number to sum
            num = 0;    // Reset for next number
        }
    }
    
    sum += num; // Add last number (if any)
    return sum;
}

// Driver code
int main() {
    string input;
    cout << "Enter a string: ";
    cin >> input;

    cout << "Sum of numbers in the string: " << sumOfNumbersInString(input) << endl;
    return 0;
}
